FRC-Extension/RoboRIOCode/RoboRIOConnection.cs

```diff
@@ -123,118 +243,139 @@ public static class RoboRIOConnection
         }
 
-        public static async Task<Dictionary<string, SshCommand>> RunCommandsAsync(string[] commands, ConnectionUser user)
+        public async Task<Dictionary<string, SshCommand>> RunCommandsAsync(IList<string> commands,
+            ConnectionUser user)
         {
-            ConnectionInfo connectionInfo;
+            SshClient ssh;
             switch (user)
             {
                 case ConnectionUser.Admin:
-                    connectionInfo = s_adminConnectionInfo;
+                    ssh = m_sshAdminClient;
                     break;
                 case ConnectionUser.LvUser:
-                    connectionInfo = s_lvUserConnectionInfo;
+                    ssh = m_sshUserClient;
                     break;
                 default:
                     throw new ArgumentOutOfRangeException(nameof(user), user, null);
             }
 
             Dictionary<string, SshCommand> retCommands = new Dictionary<string, SshCommand>();
-            using (SshClient ssh = new SshClient(connectionInfo))
+
+            var settings = SettingsProvider.ExtensionSettingsPage;
+            bool verbose = settings.Verbose || settings.DebugMode;
+            foreach (string s in commands)
             {
-                try
-                {
-                    await Task.Run(() => ssh.Connect()).ConfigureAwait(false);
-                }
-                catch (SshOperationTimeoutException)
+                if (verbose)
                 {
-                    return null;
+                    await OutputWriter.Instance.WriteLineAsync($"Running command: {s}").ConfigureAwait(false);
                 }
-                var settings = SettingsProvider.ExtensionSettingsPage;
-                bool verbose = settings.Verbose || settings.DebugMode;
-                foreach (string s in commands)
+                await Task.Run(() =>
                 {
-                    if (verbose)
-                    {
-                        await OutputWriter.Instance.WriteLineAsync($"Running command: {s}").ConfigureAwait(false);
-                    }
-                    await Task.Run(() =>
-                    {
-                        var x = ssh.RunCommand(s);
+                    var x = ssh.RunCommand(s);
 
-                        retCommands.Add(s, x);
-                    }).ConfigureAwait(false);
-                }
+                    retCommands.Add(s, x);
+                }).ConfigureAwait(false);
             }
             return retCommands;
         }
 
-        public static async Task<SshCommand> RunCommandAsync(string command, ConnectionUser user)
+        public async Task<SshCommand> RunCommandAsync(string command, ConnectionUser user)
         {
-            ConnectionInfo connectionInfo;
+            SshClient ssh;
             switch (user)
             {
                 case ConnectionUser.Admin:
-                    connectionInfo = s_adminConnectionInfo;
+                    ssh = m_sshAdminClient;
```

Comment:
Well, it's in CPS, so I can't send you the code unfortunately. Once we release the new project system and I bug you into rewriting for that, then you'll be able to use it :).